api/ocm/extensions/blobhandler/handlers/generic/ocirepo/blobhandler.go

```diff
@@ -135,27 +135,44 @@ func (b *artifactHandler) StoreBlob(blob cpi.BlobAccess, artType, hint string, g
 		tag = ":" + tag
 	}
 
+	host, port := base.HostPort()
+	artRef, err := GenOciRef(host, port, tag, version, namespace.GetNamespace())
+	if err != nil {
+		// if this not an official OCI reference, we still allow it (for legacy reasons)
+		// but return it as an OCM OCI Artifact Reference (note that this is technically not an OCI reference and invalid)
+		return ociartifact.New(base.ComposeRef(namespace.GetNamespace() + tag + version)), nil
+	}
+
+	return ociartifact.New(artRef), nil
+}
+
+func GenOciRef(host, port, tag, version, namespace string) (string, error) {
 	suffix := tag
 	if version != "" {
 		suffix += version
 	}
 
-	host, port := base.HostPort()
 	repoRef := host
 	if port != "" {
 		repoRef += ":" + port
 	}
-	if namespace.GetNamespace() != "" {
-		repoRef += "/" + namespace.GetNamespace()
+	if namespace != "" {
+		repoRef += "/" + namespace
+	} else {
+		// TODO(ikhandamirov): remove this workaround once the oci.ParseArt() function is fixed.
+		return "", errors.ErrInvalid("namespace cannot be empty in an OCI reference")
 	}
 	artRaw := repoRef + suffix
 
+	// TODO(ikhandamirov): oci.ParseArt() as a validating function has certain shortcomings.
+	// 1. If a port number is specified, it returns an error, even though the port is allowed in an OCI reference.
```

Comment:
instead of using oci.ParseArt you may want to use the oras tooling here for proper ref parsing